### PR TITLE
Parse sell/buy amount from URL and apply to Limit orders widget

### PR DESCRIPTION
**Pull request by shoom3301**
# Summary

Parse `sell/buy` amount from `URL` and apply to `Limit orders` widget
Example:
`/#/1/limit-orders/WETH/COW?sellAmount=4&buyAmount=360000`

In case when both `sellAmount` and `buyAmount` are specified, the `price` will be automatically calculated.
Warning! Amounts will be removed from the URL only when input currency or output currency is set. Because we need to know a currency for amount parsing.

### CODE REVIEW

File: src/cow-react/utils/getIntOrFloat.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+// Allow only int or float
+// Examples: 5 or 5.4 or 1. or 0.333 or .12
+export function getIntOrFloat(amount: string | null): string | null {
+  if (!amount) return null
+
+  if (/^\.\d+|\d+\.?\d*$/.test(amount)) return amount
```
Comment (anxolin) [inline src/cow-react/utils/getIntOrFloat.ts:6]: Probably `/^\d+\.?\d*$/`  is enough.

Otherwise, this seems simpler:
`/^\d+(?:\.\d*)?$`

**Review by anxolin — APPROVED** [review]
Works great!

File: src/cow-react/utils/getIntOrFloat.ts (hunk 1 above)
Comment (anxolin) [reply src/cow-react/utils/getIntOrFloat.ts:6]: aha! makes sense!

**Comment by elena-zh** [comment]
Hey @shoom3301 , when I past this link https://swap-dev-git-limit-orders-amounts-from-url-cowswap.vercel.app/#/1/limit-orders/WETH/COW?buyAmount=4, I get a sell order with a sell amount of 4. 
I expect to see there a buy order
image

The same is with the Swap page: https://swap-dev-git-limit-orders-amounts-from-url-cowswap.vercel.app/#/1/swap/WETH/COW?buyAmount=1

**Comment by elena-zh** [comment]
Amounts (parameters) are missing when change a network indicator: /#/5/limit-orders/WETH/COW?sellAmount=4&buyAmount=360000
image

**Review by nenadV91 — APPROVED** [review]

**Comment by shoom3301** [comment]
@elena-zh thanks! Fixed

File: src/cow-react/modules/limitOrders/updaters/ActiveRateUpdater/index.ts [hunk 2]
```diff
@@ -1,54 +0,0 @@
...
-      const field = orderKind === OrderKind.SELL ? 'outputCurrencyAmount' : 'inputCurrencyAmount'
-
-      updateLimitOrdersState({
-        [field]: null,
-      })
-    }
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [activeRate])
-
-  useLayoutEffect(() => {
```
Comment (nenadV91) [inline src/cow-react/modules/limitOrders/updaters/ActiveRateUpdater/index.ts:46]: Do we still handle this somewhere?

File: src/cow-react/modules/limitOrders/hooks/useUpdateCurrencyAmount.ts [hunk 3]
```diff
@@ -3,61 +3,44 @@ import { useUpdateAtom } from 'jotai/utils'
...
-        update.outputCurrencyAmount = FractionUtils.serializeFractionToJSON(outputCurrencyAmount)
-
-        // Update order type only if keeOrderKind param is not true
-        if (!keepOrderKind) {
-          update.orderKind = OrderKind.BUY
-        }
+      const { activeRate, amount, orderKind } = params
+      const field = orderKind === OrderKind.SELL ? Field.INPUT : Field.OUTPUT
+
+      const calculatedAmount = calculateAmountForRate({
```
Comment (nenadV91) [inline src/cow-react/modules/limitOrders/hooks/useUpdateCurrencyAmount.ts:28]: One benefit of doing this in a hook is that we don't need to pass all these params and they are all gathered in a hook.

File: src/cow-react/modules/limitOrders/updaters/ActiveRateUpdater/index.ts (hunk 2 above)
Comment (shoom3301) [reply src/cow-react/modules/limitOrders/updaters/ActiveRateUpdater/index.ts:46]: Yes, in the `useUpdateActiveRate` hook

**Review by elena-zh — COMMENTED** [review]
Hey @shoom3301 , great job!
Just a nitpick:

> Warning! Amounts will be removed from the URL only when input currency or output currency is set. Because we need to know a currency for amount parsing.

This rule is not working, as an amount remains to be displayed in the URL:
image
image

Thanks!

**Comment by shoom3301** [comment]
@elena-zh on your screenshot I see only one asset - COW, so, the rule is working

**Comment by elena-zh** [comment]
> @elena-zh on your screenshot I see only one asset - COW, so, the rule is working

Hey @shoom3301 , on my screenshot I set `Sell `asset and a `buy amount`, and expect that `buy amount` will be removed as a buy token is not specified. 
And vice versa: I set a 'Buy' token and a `sell amount`, and expect here  `sell amount` to be removed from the URL. 

So here is the issue I report according to your note `Amounts will be removed from the URL only when input currency or output currency is set. `

**Comment by shoom3301** [comment]
@elena-zh yes, it's right.
> Warning! Amounts will be removed from the URL only when input currency or output currency is set. Because we need to know a currency for amount parsing.

We can clean amounts from the URL, only both assets are known. `COW/ETH` for example

**Pull request merged**